homeassistant/components/touchlinesl/climate.py

```diff
@@ -0,0 +1,119 @@
+"""Roth Touchline SL climate integration implementation for Home Assistant."""
+
+from typing import Any
+
+from pytouchlinesl import Zone
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import TouchlineSLConfigEntry
+from .const import DOMAIN
+from .coordinator import TouchlineSLModuleCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: TouchlineSLConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Touchline devices."""
+    coordinators = entry.runtime_data
+    for module_id in coordinators:
+        async_add_entities(
+            TouchlineSLZone(coordinator=coordinators[module_id], zone_id=z)
+            for z in coordinators[module_id].data.zones
+        )
+
+
+CONST_TEMP_PRESET_NAME = "Constant Temperature"
+
+
+class TouchlineSLZone(CoordinatorEntity[TouchlineSLModuleCoordinator], ClimateEntity):
+    """Roth Touchline SL Zone."""
+
+    _attr_has_entity_name = True
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_mode = HVACMode.HEAT
+    _attr_hvac_modes = [HVACMode.HEAT]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+
+    def __init__(self, coordinator: TouchlineSLModuleCoordinator, zone_id: int) -> None:
+        """Construct a Touchline SL climate zone."""
+        super().__init__(coordinator, context=zone_id)
+        self.zone_id: int = zone_id
+
+        self._attr_name = None
+        self._attr_unique_id = f"{self.zone_id}"
```

Comment:
```suggestion
        self._attr_unique_id = str(zone_id)
```
This is an unique_id per zone?

Reply:
They are, but I don't know that they're unique across multiple modules so I've updated to include the module ID too.